### DISCUSSION TITLE
Is there a recommended way of implementing repeatable fields in an Rsuite form?

### DISCUSSION THREAD

**Discussion by JonathanHolvey** [rsuite/rsuite#2962] [General]
I'm trying to implement a form with a group of repeatable fields - a fairly common UX pattern, where a user can add and remove one or more items in a form, each with the same set of fields. Something like this:

![Repeatable field](https://s3.amazonaws.com/fp.strategy11.com/images/knowledgebase/repeatable-section_frontend.gif)

Is there a way this can be achieved with a state-controlled form using the Rsuite `Form` component? I've found that the way the form's `formValue`  gets mapped into the `Form.Control` components makes it very difficult to dynamically add and remove fields.

**Comment by simonguo** [comment]
Hi @JonathanHolvey  
The following is a dynamic form with data validation, I hope it can provide you with reference.
https://rsuitejs.com/components/form-validation/#dynamic-form-validation

**Reply by JonathanHolvey** [reply]
Thanks for pointing that out! I'm not sure how I missed it in the docs. Thanks again
